src/components/lab/thermometer/ThermometerShape.js

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Shape } from 'react-konva';
+import {
+  THERMOMETER_RADIUS,
+  THERMOMETER_HEIGHT,
+  THERMOMETER_WIDTH,
+  THERMOMETER_POSITION_X,
+} from '../../../config/constants';
+
+const ThermometerShape = ({
+  stroke,
+  strokeWidth,
+  fillColor,
+  thermometerHeight,
+  fillHeight,
+  offsetY,
+}) => {
+  // use trigonometry to get the exact angle matching the circle and the rectangle
+  // the angle is used to draw the most perfect mercury bulb
+  const angle = Math.asin(THERMOMETER_WIDTH / 2 / THERMOMETER_RADIUS);
+
+  const drawThermometerShape = (context, shape) => {
+    const totalOffset = offsetY + thermometerHeight - fillHeight;
+
+    context.beginPath();
+    // draw top and right straight lines
+    context.lineTo(THERMOMETER_POSITION_X, totalOffset);
+    context.lineTo(THERMOMETER_POSITION_X + THERMOMETER_WIDTH, totalOffset);
+    context.lineTo(
+      THERMOMETER_POSITION_X + THERMOMETER_WIDTH,
+      totalOffset + fillHeight,
+    );
+
+    // draw bulb
+    context.arc(
+      THERMOMETER_POSITION_X + THERMOMETER_WIDTH / 2,
+      totalOffset + fillHeight + THERMOMETER_RADIUS,
+      THERMOMETER_RADIUS,
```

Comment:
Maybe following two could be helper functions so that their role is explained in the name?